sympy/core/sympify.py

```diff
@@ -266,6 +270,23 @@ def sympify(a, locals=None, convert_xor=True, strict=False, rational=False,
     except AttributeError:
         pass
 
+    if np is not None:
+        if isinstance(a, np.ndarray):
+            #Delayed import to avoid circular imports
+            from sympy.tensor.array.dense_ndim_array import ImmutableDenseNDimArray
+            from sympy.utilities.iterables import flatten
+            try:
+                shape = a.shape
+                if len(a.shape) == 1:
+                    shape = (1,) + shape
+                a = flatten(a)
+                return ImmutableDenseNDimArray([sympify(x, locals=locals,
```

Comment:
No need to `sympify`, it is done by _ImmutableDenseNDimArray_ (though without all those parameters).